packages/jest-haste-map/src/crawlers/node.ts

```diff
@@ -71,7 +71,9 @@ function find(
     fs.readdir(directory, {withFileTypes: true}, (err, entries) => {
       activeCalls--;
       if (err) {
-        callback(result);
+        if (activeCalls === 0) {
```

Comment:
I _think_this makes sense. is the error you're seeing mitigated by this change, or is it an educated guess?

Reply:
It _is_ mitigated. I was running Jest in a debugger in our monorepo to understand what was going on and this change reliably restored operations to give stable results.

Here is an example of 2 consecutive runs in our current monorepo with the code as-is:
```shell
$ yarn jest thisdoesnotexist
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /home/oliver/fairmanager/core-web3
  6541 files checked.
  testMatch:  - 0 matches
  testPathIgnorePatterns: /node_modules/ - 6541 matches
  testRegex: (/__tests__/.*|(\.|/)(test|spec))\.tsx?$ - 172 matches
Pattern: thisdoesnotexist - 0 matches

$ yarn jest thisdoesnotexist
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /home/oliver/fairmanager/core-web3
  5436 files checked.
  testMatch:  - 0 matches
  testPathIgnorePatterns: /node_modules/ - 5436 matches
  testRegex: (/__tests__/.*|(\.|/)(test|spec))\.tsx?$ - 172 matches
Pattern: thisdoesnotexist - 0 matches
```

Not that the number of `files checked` decreased. This is the result of the map being generated, hitting the error, and then persisting the partial map to disk. Operations then continue, the search finally completes and the full set of files is pushed into the map.
When I run the second time, I only get the files that were in the cache from the first right.

After making the change suggested in this PR, the file count will restore to the original, full count:
```shell
vi node_modules/jest-haste-map/build/crawlers/node.js

$ yarn jest thisdoesnotexist
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /home/oliver/fairmanager/core-web3
  6541 files checked.
  testMatch:  - 0 matches
  testPathIgnorePatterns: /node_modules/ - 6541 matches
  testRegex: (/__tests__/.*|(\.|/)(test|spec))\.tsx?$ - 172 matches
Pattern: thisdoesnotexist - 0 matches
```

All runs after that contain the correct amount of files.

Reply:
cool, good enough for me! 😀